adapters/adocean/adocean.go

```diff
@@ -147,43 +169,52 @@ func (a *AdOceanAdapter) makeRequest(existingRequests []*adapters.RequestData, i
 		headers.Add("Referer", request.Site.Page)
 	}
 
-	return &adapters.RequestData{
-		Method:  "GET",
-		Uri:     url,
-		Headers: headers,
-	}, nil
+	requestsData = append(requestsData, &requestData{
+		Url:        url,
+		Headers:    &headers,
+		SlaveSizes: slaveSizes,
+	})
+
+	return requestsData, nil
 }
 
-func addToExistingRequest(existingRequests []*adapters.RequestData, newParams *openrtb_ext.ExtImpAdOcean, auctionID string) bool {
-requestsLoop:
-	for _, request := range existingRequests {
-		endpointURL, _ := url.Parse(request.Uri)
-		queryParams := endpointURL.Query()
+func addToExistingRequest(requestsData []*requestData, newParams *openrtb_ext.ExtImpAdOcean, imp *openrtb.Imp, testImp bool) bool {
+	auctionID := imp.ID
+
+	for _, requestData := range requestsData {
+		endpointUrl, _ := url.Parse(requestData.Url)
```

Comment:
The `makeURL` method is building a native Url object and converting it into a string. The first thing we're doing here is parsing it from a string to a Url object. Can you change `makeURL` to return a Url object and avoid this rework here?

Reply:
It's changed.